mantidimaging/gui/widgets/mi_image_view/view.py

```diff
@@ -204,17 +203,30 @@ def _update_roi_region_avg(self) -> Optional[SensibleROI]:
         # image indices are in order [Z, X, Y]
         left, right = roi_pos.x, roi_pos.x + roi_size.x
         top, bottom = roi_pos.y, roi_pos.y + roi_size.y
-        data = self.image[:, top:bottom, left:right]
-        if data is not None:
-            while data.ndim > 1:
-                data = data.mean(axis=1)
-            if len(self.roiCurves) == 0:
-                self.roiCurves.append(self.ui.roiPlot.plot())
-            self.roiCurves[0].setData(y=data, x=self.tVals)
-        self.roiString = f"({left}, {top}, {right}, {bottom}) | " \
-                         f"region avg={data[int(self.timeLine.value())].mean():.6f}"
+
+        if self.roi.isVisible():
+            z_value = int(self.timeLine.value())
+            mean_val = self.image[z_value, top:bottom, left:right].mean()
+            self.roiString = f"({left}, {top}, {right}, {bottom}) | " \
+                             f"region avg={mean_val:.6f}"
+
+        if not self.ui.roiBtn.isChecked():
+            return None
+
+        data = self.image[:, top:bottom, left:right].mean(axis=(1, 2))
+
+        if len(self.roiCurves) == 0:
+            self.roiCurves.append(self.ui.roiPlot.plot())
+        self.roiCurves[0].setData(y=data, x=self.tVals)
+
         return SensibleROI(left, top, right, bottom)
 
+    def roiClicked(self):
```

Comment:
I would be tempted to add a docstring here to explain why we need this. If i saw this in a year's time with no context, i might question why it is needed.